### PR TITLE
HAWQ-1379. Do not send options multiple times in build_startup_packet()

### PR DESCRIPTION
**Pull request by paul-guo-**
This patch also tweak the related code a bit.

### CODE REVIEW

File: src/backend/gp_libpq_fe/fe-protocol3.c [hunk 1]
```diff
@@ -2296,20 +2261,8 @@ build_startup_packet(const PGconn *conn, char *packet,
...
-				packet_len += strlen(val) + 1;
-			}
-		}		if (packet)
-			strcpy(packet + packet_len, "options");
-		packet_len += strlen("options") + 1;
-		if (packet)
-			strcpy(packet + packet_len, conn->pgoptions);
-		packet_len += strlen(conn->pgoptions) + 1;
+				ADD_STARTUP_OPTION(next_eo->pgName, val);
+		}
```
Comment (ictmalili) [inline src/backend/gp_libpq_fe/fe-protocol3.c]: There were two pieces for the same code in original code, in the PR it only has 1. Will there be a problem?

**Comment by huor** [comment]
LGTM +1

**Comment by ictmalili** [comment]
LGTM. +1

**Comment by paul-guo-** [comment]
Merged. Closing.

**Pull request closed without merging**